paddle/fluid/framework/details/multi_devices_graph_pass.cc

```diff
@@ -226,6 +237,52 @@ std::unique_ptr<ir::Graph> MultiDevSSAGraphBuilderBase::ApplyImpl(
    * Only variables should be the leaves of graph.
    */
   AddOutputToLeafOps(&result);
+
+  /*
+   * Init Gradients and FusedVars
+   */
+  if (graph->Has(kFusedVars)) {
+    VLOG(10) << "Init FusedVars.";
+    auto &fused_vars = result.Get<FusedVars>(kFusedVars);
+    for (size_t i = 0; i < local_scopes_.size(); ++i) {
+      for (auto &var : fused_vars) {
+        auto iter = all_vars_.find(var);
+        PADDLE_ENFORCE(iter == all_vars_.end());
+        local_scopes_[i]->Var(var)->GetMutable<framework::LoDTensor>();
+      }
+    }
+  }
+
+  if (graph->Has(kParamsAndGrads)) {
+    VLOG(10) << "Init Gradients.";
+    auto &params_grads = graph->Get<ParamsAndGrads>(kParamsAndGrads);
+    for (size_t i = 0; i < local_scopes_.size(); ++i) {
+      for (auto &p_g : params_grads) {
+        auto iter = all_vars_.find(p_g.second);
+        PADDLE_ENFORCE(iter != all_vars_.end());
+        PADDLE_ENFORCE_EQ(iter->second->GetType(), proto::VarType::LOD_TENSOR);
+        local_scopes_[i]->Var(p_g.second)->GetMutable<framework::LoDTensor>();
+      }
+    }
+  }
+
+  /*
+   * Run Only Once Programs
+   */
+  if (graph->Has(kRunOnlyOnceProgram)) {
+    auto &programs = graph->Get<RunOnlyOnceProgram>(kRunOnlyOnceProgram);
+    for (auto &program : programs) {
+      for (size_t i = 0; i < local_scopes_.size(); ++i) {
+        for (auto &op_desc : program.Block(0).AllOps()) {
+          auto op = paddle::framework::OpRegistry::CreateOp(*op_desc);
+          VLOG(4) << op->DebugStringEx(local_scopes_[i]);
+          op->Run(*local_scopes_[i], places_[i]);
```

Comment:
It's better to distinguish compile time and runtime, not to run anything in passes.